service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/HearingOrderKindEventDataBuilderTest.java

```diff
@@ -0,0 +1,296 @@
+    private static final DocumentReference DOCUMENT_REFERENCE_8 = mock(DocumentReference.class);
+    private static final DocumentReference DOCUMENT_REFERENCE_9 = mock(DocumentReference.class);
+    private static final UUID UUID_1 = UUID.randomUUID();
+
+    private final IdentityService identityService = mock(IdentityService.class);
+
+    private final HearingOrderKindEventDataBuilder underTest = new HearingOrderKindEventDataBuilder(identityService,
+        new OptionCountBuilder());
+
+    @Test
+    void testEmptyHearingOrderBundleDraft() {
+        when(identityService.generateId()).thenReturn(UUID_1);
+
+        UploadDraftOrdersData.UploadDraftOrdersDataBuilder newEventDataBuilder = UploadDraftOrdersData.builder();
+
+        underTest.build(SELECTED_HEARING_ID, CaseData.builder().build(), UploadDraftOrdersData.builder().build(),
+            newEventDataBuilder);
+
+        assertThat(newEventDataBuilder.build()).isEqualTo(UploadDraftOrdersData.builder()
+            .currentHearingOrderDrafts(List.of(element(UUID_1, HearingOrder.builder().build())))
+            .orderToSendOptionCount("0")
+            .orderToSend0(null)
+            .orderToSend1(null)
+            .orderToSend2(null)
+            .orderToSend3(null)
+            .orderToSend4(null)
+            .orderToSend5(null)
+            .orderToSend6(null)
+            .orderToSend7(null)
+            .orderToSend8(null)
+            .orderToSend9(null)
+            .build()
+        );
+    }
+
+    @Test
+    void testExistingHearingOrderBundleDraftWitC21() {
+
+        UploadDraftOrdersData.UploadDraftOrdersDataBuilder newEventDataBuilder = UploadDraftOrdersData.builder();
+
+        Element<HearingOrder> c21 = element(HearingOrder.builder()
+            .order(DOCUMENT_REFERENCE_0)
+            .type(HearingOrderType.C21)
+            .build());
+
+        underTest.build(SELECTED_HEARING_ID, CaseData.builder()
+                .hearingOrdersBundlesDrafts(List.of(element(UUID_1, HearingOrdersBundle.builder()
+                    .hearingId(SELECTED_HEARING_ID)
+                    .orders(Lists.newArrayList(c21))
+                    .build())))
+                .build(), UploadDraftOrdersData.builder().build(),
+            newEventDataBuilder);
+
+        assertThat(newEventDataBuilder.build()).isEqualTo(UploadDraftOrdersData.builder()
+            .currentHearingOrderDrafts(Lists.newArrayList(c21))
+            .orderToSendOptionCount("0")
+            .orderToSend0(DOCUMENT_REFERENCE_0)
+            .orderToSend1(null)
+            .orderToSend2(null)
+            .orderToSend3(null)
+            .orderToSend4(null)
+            .orderToSend5(null)
+            .orderToSend6(null)
+            .orderToSend7(null)
+            .orderToSend8(null)
+            .orderToSend9(null)
+            .build()
+        );
+    }
+
+    @Test
+    void testExistingHearingOrderBundleDraftWitMultipleC21() {
```

Comment:
Similar to above, change from "testExistingHearingOrderBundleDraftWitMultipleC21" to "testExistingHearingOrderBundleDraftWithMultipleC21"

Reply:
👍